### ISSUE TITLE
Not able to use the predict function properly when loaded the saved h5 model

### ISSUE THREAD

**Issue by JW-Wayne-Zheng** [WayneJWZLemon/FlaskGANDemo#1]
```python
model = keras.models.load_model("models/monet_generator.h5", custom_objects={"InstanceNormalization":tfa.layers.InstanceNormalization})
img = cv2.imread("images/00068bc07f.jpg")
print(img.shape)
img = reshape(img, shape=[1, 256, 256, 3])
print(img.shape)
prediction = model(img, training=False)[0].numpy() # make predition
prediction = (prediction * 127.5 + 127.5).astype(np.uint8)   # re-scale
im = PIL.Image.fromarray(prediction)
im.save(f'{"output_images"}.jpg')
```
The above code did allow us to load a model from the saved h5 file, however, the predict function always produces a white image (256x256).

**Comment by JW-Wayne-Zheng** [comment]
As suggested by my team member Conor, it is probably worth trying out to first have the original generator class (in the Kaggle Notebook) defined in the Flask app, then use the load_weights() on the h5 file.

**Comment by JW-Wayne-Zheng** [comment]
```python
#!/usr/bin/env python
# coding: utf-8

# In[12]:


# example of using saved cyclegan models for image translation
#https://machinelearningmastery.com/cyclegan-tutorial-with-keras/
from numpy import load
from numpy import expand_dims
from keras.models import load_model
from keras_contrib.layers.normalization.instancenormalization import InstanceNormalization
from keras.preprocessing.image import img_to_array
from keras.preprocessing.image import load_img
from matplotlib import pyplot
from flask import Flask, render_template
from tensorflow.keras.models import load_model
import tensorflow_addons as tfa
from bs4 import BeautifulSoup
import cv2
import tensorflow as tf
import PIL
import numpy as np
import generator
# load an image to the preferred size
def load_resize_image(filename, size=(256,256)):
	# load and resize the image
	pixels = load_img(filename, target_size=size)
	# convert to numpy array
	pixels = img_to_array(pixels)
	# transform in a sample
	pixels = expand_dims(pixels, 0)
	# scale from [0,255] to [-1,1]
	pixels = (pixels - 127.5) / 127.5
	return pixels
 
# load the image
image_src = load_resize_image('models/0a49020ae5.jpg')
# load the model
model_AtoB = load_model('models/monet_generator.h5')
# translate image
image_tar = model_AtoB.predict(image_src)
# scale from [-1,1] to [0,1]
image_tar = (image_tar + 1) / 2.0
# plot the translated image
pyplot.imshow(image_tar[0])
pyplot.show()
app = Flask(__name__)

@app.route('/')
def index():
    return render_template("index.html")

if __name__=="__main__":
    app.run(debug=True)


# In[ ]:

```
The above code snippet did allow us to see the predicted/style-transferred image in the plot.

**Comment by JW-Wayne-Zheng** [comment]
```python
pyplot.imshow(image_tar[0])
pyplot.axis("off")
pyplot.savefig('output.jpg', bbox_inches='tight', pad_inches=0)
```
The above code allows us to save the transferred image without axis and white space using matplotlib.
